lib/src/layer/polygon_layer/label.dart

```diff
@@ -17,46 +17,38 @@ void Function(Canvas canvas)? buildLabelTextPainter(
   TextStyle? labelStyle,
   double padding = 0,
 }) {
-  double dx = placementPoint.dx;
-  double dy = placementPoint.dy;
+  final textSpan = TextSpan(text: labelText, style: labelStyle);
+  final textPainter = TextPainter(
+    text: textSpan,
+    textAlign: TextAlign.center,
+    textDirection: TextDirection.ltr,
+    maxLines: 2,
+  )..layout();
 
-  if (dx > 0) {
```

Comment:
This check is super weird, it removes labels when their center goes off screen but only to the left. It's jarring to look at.

Reply:
Wouldn't it be better to fix this to remove labels when off screen in any direction, rather than remove this optimization altogether like you appear to have done? (Unless I've missed something that is.)

Reply:
We could.

For now, I'd rather call it: "fixing visual glitches" rather than "removing an optimization that only 'work' when labels slip out to the left" :).

Computationally it would probably still be worth it. However, note that right now we're already culling polygons. So, this optimization would only kick in for polygons that are only partially on screen with the label off screen.

Would you wanna do it now or later?

Reply:
I guess that most of the time, the label is likely to be present. I guess technically 50% of the time if all locations had equal chance.
You're right, let's leave it for now. I think culling the polygons is probably enough.